pkg/controller/clone/planner.go

```diff
@@ -305,6 +321,58 @@ func (p *Planner) computeStrategyForSourcePVC(ctx context.Context, args *ChooseS
 	return &strategy, nil
 }
 
+func (p *Planner) computeStrategyForSourceSnapshot(ctx context.Context, args *ChooseStrategyArgs) (*cdiv1.CDICloneStrategy, error) {
+	if ok, err := p.validateTargetStorageClassAssignment(ctx, args); !ok || err != nil {
+		return nil, err
+	}
+
+	// Defaulting to host-assisted clone since it has fewer requirements
+	strategy := cdiv1.CloneStrategyHostAssisted
+
+	// Do volume snapshot validation before checking other specific stuff
+	sourceSnapshot := &snapshotv1.VolumeSnapshot{}
+	exists, err := getResource(ctx, p.Client, args.DataSource.Namespace, args.DataSource.Spec.Source.Name, sourceSnapshot)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		message := fmt.Sprintf(MessageCloneWithoutSource, "snapshot", args.DataSource.Spec.Source.Name)
+		p.Recorder.Event(args.TargetClaim, corev1.EventTypeWarning, CloneWithoutSource, message)
+		args.Log.V(3).Info("Source Snapshot does not exist, cannot compute strategy")
+		return nil, nil
+	}
+	valid, err := cc.IsSnapshotValidForClone(sourceSnapshot, args.Log)
+	if err != nil || !valid {
+		p.Recorder.Event(args.TargetClaim, corev1.EventTypeWarning, CloneValidationFailed, MessageCloneValidationFailed)
+		return nil, err
+	}
+
+	// Do snapshot and storage class validation
+	targetStorageClass, err := cc.GetStorageClassByName(ctx, p.Client, args.TargetClaim.Spec.StorageClassName)
```

Comment:
I don't love this function in populator context because it has some business logic we don't care about (returns default volumesnapshotclass if name is nil).  A simple get makes more sense to me but your call